### PR TITLE
[build system]: move to webpack@2, add source maps

### PR DESCRIPTION
**Pull request by legomushroom**
This PR:
- changes the `webpack` config to `v2.2.x`
- updates all dependencies to their current versions
- adds `source maps` support for `dev` and `prod` environment

Had to update all dependencies because after `npm install`, the build system was broken with strange `v8` error.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -38,16 +38,18 @@ function getBundleConfig() {
...
         );
     } else {
         config.output.pathinfo = true;
     }
 
-    if (!RELEASE || DEBUG) {
-        config.devtool = '#inline-source-map';
-    }
+    config.devtool = (RELEASE)
+                        ? 'source-map'
```
Comment (davidkpiano) [inline gulpfile.js]: Weird formatting here

Comment (legomushroom) [reply gulpfile.js]: Any suggestions how you would format it?

Comment (davidkpiano) [reply gulpfile.js]: It's not a big deal - we can definitely leave it the way it is ☺️ I'd just do a single tab indent though:

```js
config.devtool = (RELEASE)
    ? 'source-map'
    : 'cheap-module-eval-source-map';
```

Comment (legomushroom) [reply gulpfile.js]: This form is not clearer for me, I always confused where does this `?` belong to. I was able to change it to one-liner for this case.

**Pull request merged**